src/charm.py

```diff
@@ -222,6 +222,11 @@ def _on_reset_instance_action(self, event: ActionEvent, charm_state: CharmState)
             return
         try:
             self.model.unit.status = ops.MaintenanceStatus("Resetting Synapse instance")
+            try:
+                container.stop(pebble.STATS_EXPORTER_SERVICE_NAME)
+            except (ops.pebble.Error, RuntimeError) as e:
```

Comment:
Are we sure that the service was stopped if an exception is raised?

Reply:
No, but if the container dont stop, the action will fail anyway since if there are open connections in the database:
"database "synapse" is being accessed by other users"
https://github.com/canonical/synapse-operator/actions/runs/8207941891/job/22450994126

Reply:
Perhaps if we encounter an error we fail at that action at that point then?

Reply:
Good suggestion, I changed to fail the event at this point.